src/browser-lib/extra-providers.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable, inject, NgZone, StaticProvider, Inject } from '@angular/core';
+import {
+  ɵBrowserPlatformLocation,
+  PlatformLocation,
+  LocationChangeEvent,
+  DOCUMENT,
+} from '@angular/common';
+
+@Injectable()
+class SingleSpaPlatformLocation extends ɵBrowserPlatformLocation {
+  onPopState(fn: (event: LocationChangeEvent) => void): void {
+    const ngZone = inject(NgZone);
+
+    super.onPopState(event => {
+      // Wrap any event listener into zone that is specific to some application.
+      // The main issue is `back/forward` buttons of browsers, because they invoke
+      // `history.back|forward` which dispatch `popstate` event. Since `single-spa`
+      // overrides `history.replaceState` Angular's zone cannot intercept this event.
+      // Only the root zone is able to intercept all events.
```

Comment:
👍 good explanation. Let's also add a link to the github issue for those wanting to read more about it.

```js
// See https://github.com/single-spa/single-spa-angular/issues/94 for more detail
```